diktat-rules/src/test/kotlin/com/saveourtool/diktat/ruleset/chapter2/KdocMethodsTest.kt

```diff
@@ -272,6 +272,81 @@ class KdocMethodsTest : LintTestBase(::KdocMethods) {
         )
     }
 
+    @Test
+    @Tag(WarningNames.KDOC_WITHOUT_THROWS_TAG)
+    fun `No warning when catch without matching throw`() {
+        lintMethod(
+            """
+                    /**
+                      * Test method
+                      * @param a: Int - dummy integer
+                      * @return doubled value
+                      */
+                    fun foo(a: Int): Int {
+                        try {
+                            if (a < 0)
+                                throw ex1()
+                        } catch (e: ex1) {
+                            print(1)
+                        } catch (e: ex2) {
+                            print(2)
+                        } catch (e: ex3) {
+                            print(3)
+                        }
+                        return 2 * a
+                    }
+            """.trimIndent())
+    }
+
+    @Test
+    @Tag(WarningNames.KDOC_WITHOUT_THROWS_TAG)
+    fun `Warning when throw doesn't have matching catch`() {
+        lintMethod(
+            """
+                    /**
+                      * Test method
+                      * @param a: Int - dummy integer
+                      * @return doubled value
+                      */
+                    fun foo(a: Int): Int {
+                        try {
+                            if (a < 0)
+                                throw ex1()
+                        throw ex2()
+                        throw ex3()
+                        } catch (e: ex1) {
+                            print(1)
+                        } catch (e: ex2) {
+                            print(2)
+                        }
```

Comment:
Could you use real exceptions? It may be worth adding a case related to child exceptions, as in fix test. I also think it's worth putting this in the fix test to show the addition of `@throws` for the remaining exceptions.

Reply:
Done